app/controllers/units_controller.rb

```diff
@@ -67,76 +75,85 @@ def update
-      end
-      if ! ajax_modal_request?
-        redirect_to building_path(@unit.building)
+        # op = Opportunity.where(unit_id: @unit[:id]).first_or_create(unit: @unit, available: true)
       end
+      redirect_to building_path(@unit.building) unless ajax_modal_request?
       flash[:notice] = "Unit <strong>#{@unit[:name]}</strong> was successfully updated."
     else
-      render :edit, {:flash => { :error => 'Unable to update unit <strong>#{@unit[:name]}</strong>.'}}
+      render :edit, { flash: { error: "Unable to update unit <strong>#{@unit[:name]}</strong>." } }
     end
   end
 
   # DELETE /hmis/units/1
   def destroy
-    building = @unit.building
-    if @unit.destroy
-      redirect_to building_path(building), notice: "Unit <strong>#{@unit[:name]}</strong> was successfully deleted."
+    previous_page = Rails.application.routes.recognize_path(request.referrer)
+    next_url = if previous_page[:controller] == 'units' && previous_page[:action] == 'edit'
+      building_path(@unit.building)
     else
-      render :edit, {:flash => { :error => 'Unable to delete unit <strong>#{@unit[:name]}</strong>.'}}
+      request.referer
     end
-  end
 
-  # RESTORE /hmis/units/1
-  def restore
-    if Unit.restore(params[:id])
-      @unit = Unit.find(params[:id])
-      redirect_to units_path, notice: "Unit <strong>#{@unit[:name]}</strong> successfully restored."
+    if @unit.destroy
+      redirect_to next_url, notice: "Unit <strong>#{@unit[:name]}</strong> was successfully deleted."
     else
-      render :edit, {:flash => { :error => 'Unable to restore unit.'}}
+      redirect_to next_url, { flash: { error: "Unable to delete unit <strong>#{@unit[:name]}</strong>." } }
     end
   end
 
-  def get_unit
+  # POST /hmis/units/1/restore
+  def restore
     @unit = Unit.find(params[:id])
+    @unit.update(active: true)
+    redirect_back(fallback_location: units_path, notice: "Unit <strong>#{@unit[:name]}</strong> was restored.")
+  end
+
+  # POST /hmis/units/1/deactivate
+  def deactivate
+    @unit = Unit.find(params[:unit_id])
+    @unit.update(active: false)
+    redirect_back(fallback_location: units_path, notice: "Unit <strong>#{@unit[:name]}</strong> was deactivated.")
   end
 
   private
-    def unit_scope
-      Unit
-    end
-    # Use callbacks to share common setup or constraints between actions.
-    def set_unit
-      @unit = Unit.find(params[:id])
-    end
 
-    def set_building
-      if params[:building_id]
-        @unit.building = Building.find(params[:building_id])
-      end
-    end
-    # Only allow a trusted parameter "white list" through.
-    def unit_params
-      params.require(:unit).permit(
-        :name,
-        :available,
-        :building_id,
-        :elevator_accessible,
-        requirements_attributes: [:id, :rule_id, :positive, :variable, :_destroy]
-      )
-    end
+  def unit_scope
+    Unit
+  end
```

Comment:
usually this would be `unit_source` and `unit_scope` would be something like `unit_source.visible_to(current_user)`.  Given that Unit is only one word, and the scope doesn't include a scope, I'd probably just remove this method and use `Unit` instead as is being done on line 125.

Reply:
Yeah agreed. This diff is mostly rubocop fixes. I will update though